src/components/tree-item/tree-item.ts

```diff
@@ -0,0 +1,316 @@
+import { LocalizeController } from '@shoelace-style/localize';
+import { LitElement, html } from 'lit';
+import { customElement, property, query, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { live } from 'lit/directives/live.js';
+import { when } from 'lit/directives/when.js';
+import { animateTo, shimKeyframesHeightAuto, stopAnimations } from 'src/internal/animate';
+import { stringMap } from 'src/internal/string';
+import { getAnimation, setDefaultAnimation } from 'src/utilities/animation-registry';
+import { emit } from '../../internal/event';
+import { watch } from '../../internal/watch';
+import styles from './tree-item.styles';
+
+export function isTreeItem(element: Element) {
+  return element && element.getAttribute('role') === 'treeitem';
+}
+
+/**
+ * @since 2.0
+ * @status experimental
+ *
+ * @dependency sl-checkbox sl-spinner sl-icon-button
```

Comment:
This isn't obvious at all, but these need to be on separate lines otherwise all except the first will be ignored.

Reply:
Also not obvious — we need to import checkbox, spinner, and icon button here. See color picker's source for reference. https://github.com/shoelace-style/shoelace/blob/00f98cc5050e1c06374330ed28228fbf69aa92ae/src/components/color-picker/color-picker.ts#L8-L13

Reply:
fixed [6413915](https://github.com/shoelace-style/shoelace/pull/823/commits/641391589df9c9fbd8c598caf6e8e5afe21a1a71)